doc/contribute/modifying_contributions.rst

```diff
@@ -0,0 +1,70 @@
+.. _modifying_contributions:
+
+Modifying Contributions made by other developers
+************************************************
+
+Scenarios
+#########
+
+Zephyr contributors and collaborators are encouraged to assist
+as reviewers in pull requests, so that patches may be approved and merged
+to Zephyr's main branch as part of the original pull requests. The authors
+of the pull requests are responsible for amending their original commits
+following the review process.
+
+There are occasions, however, when a contributor might need to modify patches
+included in pull requests that are submitted by other Zephyr contributors.
+For instance, this is the case when:
+
+* a developer cherry-picks commits submitted by other contributors into their
+  own pull requests in order to:
+
+  * integrate useful content which is part of a stale pull request, or
+  * get content merged to the project's main branch as part of a larger
+    patch
+
+* a developer pushes to a branch or pull request opened by another
+  contributor in order to:
+
+  * assist in updating pull requests in order to get the patches merged
+    to the project's main branch
+  * drive stale pull requests to completion so they can be merged
+
+
+Accepted policies
+#################
+
+A developer who intends to cherry-pick and potentially modify patches sent by
+another contributor shall:
+
+* clarify in their pull request the reason for cherry-picking the patches,
+  instead of assisisting in getting the patches merged in their original
+  pull request, and
+* invite the original author of the patches to their pull request review.
+
+A developer who intends to force-push to a branch or pull request of
+another Zephyr contributor shall clarify in the pull request the reason
```

Comment:
Should we state that "Zephyr contributor shall clarify" _before_ doing the force-push? I.e. a user who wants to force push should first notify the author of the original PR of their intent? Just to avoid a situation when someone, suddenly, makes a force push while the original author has already made some changes localy but didn't push their work to the github repo. I assume that's the expected behavior but at the moment it is not explicit.

Reply:
@mnkp thanks for your comment: I would think this is rather intuitive; we should always be careful with force-pushing, that's why I think this might be not needed here. Besides:
- we stress that force-pushing to other's branch/PR should be limited to stale PRs, i.e. with un-responsive authors, and for reasons like to drive the PR to completion (when the author is not able to do themselves), so I would say the "race-condition" is not the use-case here :)

Reply:
"Race-condition" is not the point.  The point is who has ownership of the *branch* and so should be in control of what's in it.  The branch is in the submitter's repository.  Anybody pushing to a branch in my repos without getting a positive acknowledgement from me that it's OK is not going to be well thought of.  People may manage their branches in ways we don't understand, and a force push from other content might break that management.

My personal policy is in line with that: I'll only push if the owner of the repo authorizes it.  In any other case where I was taking over a stale PR under the conditions authorized in this policy I would issue a new PR from my repo, referencing the original which would be closed.

Reply:
> My personal policy is in line with that: I'll only push if the owner of the repo authorizes it. In any other case where I was taking over a stale PR under the conditions authorized in this policy I would issue a new PR from my repo, referencing the original which would be closed.

@pabigot although I am fully supporting more transparency and more communication in the Project (and your proposal is ideal with respect to transparency/communication), I am afraid, this may introduce more problems than what it solves.
- asking for authorization to force push: then we need to define how to do it: Slack? E-mail? GH comment? And for how long we should wait until we flag the author as non-responsive? 
- issuing a new PR from your repo is problematic: you lose comment/review history, and will need to ask around, once more, for all the approvals you had in the original PR. IMHO, this slows down a lot the process.

I assume a common case of a PR that is already reviewed, approved, but with Git conflicts, or broken CI, and the author being on holiday for a while; a trivial force-push should do it (with proper commenting, of course, as we explain here) and for the Release Engineer it would be much easier to get the PR merged, rather than looking at a brand-new PR with the same changes.

If anyone force-pushes to your branch to get CI/conflicts fixed and get it merged; I don't clearly see why this should, in general not be well thought of.

In any case, authors can always uncheck the relevant GH box to disallow adding / modifying commits